xdsl/frontend/python_code_check.py

```diff
@@ -189,18 +342,15 @@ class CheckAndInlineConstants:
     b: Const[i32] = a * a
     # here b = 25
     ```
-
-    Note that the algorithm does not remove constant definitions from the AST,
-    but this functionality can be added later.
     """
 
     @staticmethod
-    def run(stmts: Sequence[ast.stmt], file: str | None) -> None:
+    def run(stmts: list[ast.stmt], file: str | None) -> None:
```

Comment:
why this change?

Reply:
Should I keep it as a sequence? I was working on the old branch with `List[ast.stmt]` and as I can remember we wanted to have `list` instead?

Reply:
`list` is better than `List`, but Sequence is generally easier to deal with, primarily because it's covariant so accepts lists of subclasses

Reply:
Let me use a Sequence then instead.